src/posix.mak

```diff
@@ -544,8 +544,21 @@ $(optabgen_files): optabgen.out
 optabgen.out : $G/optabgen
 
 ######## VERSION
+########################################################################
+# The version file should be updated on every build
+# However, the version check script only touches the VERSION file if it
+# actually has changed.
+
+$G/version_check: ../config.d $(HOST_DMD_RUN)
+	@echo "  (HOST_DMD_RUN)  $<  $<"
+	$(HOST_DMD_RUN) $< -of$@
+
+$G/VERSION: $G/version_check ../VERSION FORCE
+	@$< $G ../VERSION $(SYSCONFDIR)
+
+$G/SYSCONFDIR.imp: $G/VERSION
```

Comment:
It probably would be nicer to do this directly in this target (would also allow us to drop the special casing from `config.d`:

```diff
diff --git a/config.d b/config.d
index 606a2b62c..a214ad9b4 100644
--- a/config.d
+++ b/config.d
@@ -6,19 +6,10 @@ void main(const string[] args)
     const outputDirectory = args[1];
     const versionFile = args[2];
 
-    version (Posix)
-        const sysConfigDirectory = args[3];
-
     mkdirRecurse(outputDirectory);
     const version_ = generateVersion(versionFile);
 
     updateIfChanged(buildPath(outputDirectory, "VERSION"), version_);
-
-    version (Posix)
-    {
-        const path = buildPath(outputDirectory, "SYSCONFDIR.imp");
-        updateIfChanged(path, sysConfigDirectory);
-    }
 }
 
 /**
diff --git a/src/posix.mak b/src/posix.mak
index 3a83236eb..724e95285 100644
--- a/src/posix.mak
+++ b/src/posix.mak
@@ -554,9 +554,10 @@ $G/version_check: ../config.d $(HOST_DMD_RUN)
 	$(HOST_DMD_RUN) $< -of$@
 
 $G/VERSION: $G/version_check ../VERSION FORCE
-	@$< $G ../VERSION $(SYSCONFDIR)
+	@$< $G ../VERSION
 
-$G/SYSCONFDIR.imp: $G/VERSION
+$G/SYSCONFDIR.imp:
+	@[ "$(cat $@)" == "$(SYSCONFDIR)" ] || echo "$(SYSCONFDIR)" > $@
 
 FORCE: ;
```

Reply:
This did not work. The `SYSCONFDIR.imp` is needed by the `dinifile.d` file which is included in the Dub package. I added back the code as it was.